charts/consul/templates/dns-proxy-deployment.yaml

```diff
@@ -1,5 +1,4 @@
 {{- if (or (and (ne (.Values.dns.proxy.enabled | toString) "-") .Values.dns.proxy.enabled) (and (eq (.Values.dns.proxy.enabled | toString) "-") .Values.global.enabled)) }}
-{{- if not .Values.connectInject.enabled }}{{ fail "connectInject.enabled must be true" }}{{ end -}}
```

Comment:
this should not have been a requirement.

Reply:
Shouldn't we have the opposite requirement? Is dns-proxy safe to use with connect-inject enabled?

Reply:
I can't think of any reason why it would be unsafe.  I think ultimately services would use the local dataplane or agent, but I can't think of a reason why it would be unsafe and require us to prevent it being configured.

Reply:
The only scenario I can think of where I'm not sure is with tproxy on. I'm not sure how we handle DNS in that case and if it's even allowed to resolve DNS at all because all traffic will be handled through envoy and virtual ips but yes probably not a big deal.